cmd/main.go

```diff
@@ -39,6 +40,21 @@ func main() {
 
 	options := GetOptions(fs)
 
+	// Start tracing as soon as possible
+	if options.ServerOptions.EnableOtelTracing {
+		exporter, err := driver.InitOtelTracing()
+		if err != nil {
+			klog.ErrorS(err, "failed to initialize otel tracing")
+			klog.FlushAndExit(klog.ExitFlushTimeout, 1)
+		}
+		// Exporter will flush traces on shutdown
+		defer func() {
+			if err := exporter.Shutdown(context.Background()); err != nil {
```

Comment:
Does `context.Background` here mean the shutdown could theoretically hang forever? Otherwise, lgtm

Reply:
Good point. I've added a 10 seconds timeout on the context, it should be plenty for the exporter to flush all remaining traces to the collector. I can probably reduce the timeout a bit if you prefer.